Fix conditions for deployment status comments

carlobeltrame opened the pull request:
Follow-up from #3202. Not sure what I was thinking before, but the conditions for the comment steps were completely wrong. Also added a status comment when a deployment fails.

carlobeltrame committed:
Fix conditions for deployment status comments

Not sure what I was thinking before, but the conditions for the comment steps were completely wrong.

BacLuc reviewed (approved):
ein secret ist wahrscheinlich nicht richtig konfiguriert.
schlimmer wird es anscheinend nicht

carlobeltrame commented:
> ein secret ist wahrscheinlich nicht richtig konfiguriert.
> schlimmer wird es anscheinend nicht

Ja das secret von den coupon codes wird von Go als Zahl interpretiert weil es nur Zahlen enthält. Braucht vermutlich noch irgendein printf vor dem b64enc. Kann ich bald noch anschauen falls es sonst niemand macht.

BacLuc commented:
Das mit den logs klappt nicht ganz, kann man aber auch später noch machen:
https://github.com/ecamp/ecamp3/pull/3223#issuecomment-1403993205

carlobeltrame committed:
Fix retrieving the job id

Also retrieve it earlier, so our `if: failure()` has a higher chance of getting all the information it needs

carlobeltrame commented:
> Das mit den logs klappt nicht ganz, kann man aber auch später noch machen: [#3223 (comment)](https://github.com/ecamp/ecamp3/pull/3223#issuecomment-1403993205)

Merci für den Hinweis und fürs demonstrieren wie man solche Workflow Änderungen direkt auf GitHub testen kann! Grund war dass GitHub den Matrix Job Namen einfach mal abschneidet. Ist gefixt:
![Screenshot 2023-01-26 at 11-51-12 Fix conditions for deployment status comments by carlobeltrame · Pull Request #3234 · ecamp_ecamp3](https://user-images.githubusercontent.com/7566995/214818126-034ec3bb-a881-491e-8449-0073707d447d.png)
